server/internal/database/job.go

```diff
@@ -106,11 +106,27 @@ func (r *JobORM) GetAllByProjectID(projectID string) ([]*models.Job, error) {
 		return jobs, nil
 	}
 
-	// Build query
+	// Extract IDs for filtering
+	sourceIDs := make([]int, len(sources))
+	for i := range sources {
+		sourceIDs[i] = sources[i].ID
+	}
+
+	destIDs := make([]int, len(destinations))
+	for i := range destinations {
+		destIDs[i] = destinations[i].ID
+	}
+
+	// Build query for jobs
 	qs := r.ormer.QueryTable(r.TableName)
-	// Filter by sources or destinations from the project
-	if len(sources) > 0 {
-		qs = qs.Filter("source_id__in", sources)
+
+	// Create OR condition for sources and destinations
+	cond := orm.NewCondition()
+	if len(sourceIDs) > 0 {
+		cond = cond.Or("source_id__in", sourceIDs)
+	}
+	if len(destIDs) > 0 {
+		cond = cond.Or("dest_id__in", destIDs)
 	}
 
 	if len(destinations) > 0 {
```

Comment:
I think the below code is not required
```suggestion
	if len(destinations) > 0 {
		qs = qs.Filter("dest_id__in", destinations)
	}
```

Reply:
removed